packages/terra-list/src/MultiSelectList.jsx

```diff
@@ -25,40 +29,65 @@ const propTypes = {
 const defaultProps = {
   children: [],
   isDivided: false,
+  hasSections: false,
   onChange: undefined,
   maxSelectionCount: undefined,
 };
 
 class MultiSelectList extends React.Component {
 
+  static getDerivedStateFromProps(props, state) {
+    if (props.hasSections !== state.hasSections) {
+      return {
+        hasSections: props.hasSections,
```

Comment:
Maybe I'm missing something, but why does hasSections need to be stored as state?

Reply:
I initially made the component to load when the inital render happens based on props. But @bjankord  has suggested that, props might change depending upon different use-cases and this should be handled with getDerivedStateFromProps static func.

More about [getDerivedStateFromProps](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops) and [StackOverFlow](https://stackoverflow.com/questions/49617486/how-to-use-lifecycle-method-getderivedstatefromprops-as-opposed-to-componentwill)